spec/app.spec.js

```diff
@@ -79,6 +102,114 @@ describe('app.spec: app namespace tests', () => {
+                try {
+                    for (let i = 0; i < 1000; i++) {
+                        Neutralino.app.broadcast('testEvent' + i);
+                    }
+                    await __close('success');
+                } catch (err) {
+                await __close('failed');
+            }
+            `);
+            assert.equal(runner.getOutput(), 'success');
+        });
+
+        it('triggers all registered listeners when broadcasting an event', async () => {
+            runner.run(`
+                let listener1Triggered = false;
+                let listener2Triggered = false;
+            
+                function listener1() {
+                    listener1Triggered = true;
+                }
+                
+                function listener2() {
+                    listener2Triggered = true;
+                    __close(JSON.stringify({ listener1Triggered, listener2Triggered }));
+                }
+            
+                await Neutralino.events.on('multiListenerEvent', listener1);
+                await Neutralino.events.on('multiListenerEvent', listener2);
+            
+                await Neutralino.app.broadcast('multiListenerEvent');
+            `);
+
+            const output = JSON.parse(runner.getOutput());
+            const bothListenersTriggered = output.listener1Triggered && output.listener2Triggered
+            assert.strictEqual(bothListenersTriggered, true, 'Both listerners should be triggered');
+        });
+
+        it('handles errors in listeners gracefully', async () => {
+            runner.run(`
+                let safeListenerCalled = false;
+            
+                function errorListener() {
+                    throw new Error('Test error');
+                }
+            
+                function safeListener() {
+                    safeListenerCalled = true;
+                    __close(JSON.stringify({safeListenerCalled}))
+                }
+            
+                await Neutralino.events.on('errorEvent', errorListener);
+                await Neutralino.events.on('errorEvent', safeListener);
+            
+                await Neutralino.app.broadcast('errorEvent');
+            `);
+            const output = JSON.parse(runner.getOutput())
+            assert.strictEqual(output.safeListenerCalled, true, 'Safe listener should be called');
+        });
+    
+        it('executes listeners in the order they were added', async () => {
+            runner.run(`
+                let order = [];
+            
+                function listener1() {
+                    order.push(1);
+                }
+            
+                function listener2() {
+                    order.push(2);
+                }
+            
+                await Neutralino.events.on('orderEvent', listener1);
+                await Neutralino.events.on('orderEvent', listener2);
+            
+                await Neutralino.app.broadcast('orderEvent');
```

Comment:
Thanks for updating the code according to notes. Btw, seems like here also we need to do the fix?